src/quo/components/wallet/required_tokens/view.cljs

```diff
@@ -0,0 +1,66 @@
+(ns quo.components.wallet.required-tokens.view
+  (:require [quo.components.markdown.text :as text]
+            [quo.components.utilities.token.view :as token]
+            [quo.components.wallet.required-tokens.style :as style]
+            quo.theme
+            [react-native.core :as rn]
+            [schema.core :as schema]))
+
+(def ?schema
+  [:=>
+   [:catn
+    [:props
+     [:map {:closed true}
+      [:type {:optional true} [:enum :token :collectible]]
+      [:amount {:optional true} [:maybe [:or :string :int]]]
+      [:token {:optional true} [:maybe :string]]
+      [:token-img-src {:optional true} [:maybe :schema.common/image-source]]
+      [:collectible-img-src {:optional true} :schema.common/image-source]
+      [:collectible-name {:optional true} [:maybe :string]]
+      [:divider? {:optional true} [:maybe :boolean]]
+      [:theme :schema.common/theme]
+      [:container-style {:optional true} [:maybe :map]]]]]
+   :any])
+
+(defn- view-internal
+  [{:keys [type amount token token-img-src collectible-img-src collectible-name divider? theme
+           container-style]
+    :or   {type :token}}]
+  [rn/view
+   {:style               (merge style/container container-style)
+    :accessibility-label :wallet-required-tokens}
+   (case type
+     :token       [token/view
+                   (merge {:size  16
+                           :token token}
+                          (when token-img-src
+                            {:image-source token-img-src}))]
```

Comment:
Just a small comment.

I'm not a big fan of using `merge` to just add one key to a map 🤔, actually, if we just want to add keys to a map,`assoc` is enough.

since we only need `:token` or `:image-source` (not both), maybe:

```clojure
(assoc (if token-img-source
        {:image-source token-img-src}
        {:token token}
  :size 16)
```

It's less readable. Maybe extracting the map to a let binding will improve it.

Is up to you, not too important, just wanted to share it.

Reply:
@ulisesmac We always need to pass the `:token`, even when we provide the `:image-source` prop because the `:token` is utilized to render the temporary loading state. Using `assoc`, the following is the best I could come up with.

```cljs
(let [base-map {:size 16 :token token}]
  (if token-img-src
    (assoc base-map :image-source token-img-src)
    base-map))
```

However, in my view, the prior implementation using `merge` is more readable. While `assoc` is generally considered more performant than `merge`, the difference may not be significant in this case, I'm not sure tho. What do you suggest?'

Reply:
Upon reflection, `assoc` seems more readable to me. I've updated the code accordingly. Please review, and I apologize for any confusion.

Reply:
> the :token is utilized to render the temporary loading state.
  
AFAIK the token component only takes one of both, image-source or the token name. Also that component doesn't have a loading state :thinking:  
I'm a little confused :sweat_smile: 

Anyway, if we need both and just `token-img-src` is conditionally added,  consider also using `cond->`:

```clojure
(cond-> {:size 16
         :token token}
  token-img-src (assoc :image-source token-img-src)
```

I have mixed feelings about it, since it's less performant, sometimes it's so hard to find a nice solution :cry:
As I said before, this is not too important :) pick whatever you think fits better :+1: :+1:

Reply:
When I sought improvement, Copliot suggested the above approach! 💪

Reply:
> AFAIK the token component only takes one of both, image-source or the token name. Also that component doesn't have a loading state 🤔

You are right, I just confused [this part](https://github.com/status-im/status-mobile/blob/ed53fecf47d52ee17941e87a52a90d0e1467c7ba/src/quo/components/utilities/token/view.cljs#L39-L49), but its under a `if`. So we can pass `:token` or `:image-source`. I have updated the code accordingly.

Thank you!